scripts/metadata_comparison/test/test_digester.py

```diff
@@ -31,89 +25,162 @@ def test_digestion(self) -> None:
         runs so much more quickly. GCS testing can be turned off by setting the DIGESTER_TEST_LOCAL_ONLY environment
         variable.
         """
-        bucket_name = 'papi-performance-analysis'
-        subdir = 'exome_germline_single_sample_v1.3/PAPIv2_alpha1/v1_style_machine_types'
-
-        local_parent = ComparisonPath.create(subdir)
-        gcs_parent = ComparisonPath.create(f'gs://{bucket_name}/{subdir}')
 
         credentials, project_id = google.auth.default()
         storage_client = storage.Client(credentials=credentials)
+
+        bucket_name = 'papi-performance-analysis'
         bucket = storage_client.get_bucket(bucket_name)
 
-        def download_metadata_from_gcs(_local_sample_path: ComparisonPath) -> None:
-            _local_sample_path.mkdir_p()
-            (_local_sample_path / "operations").mkdir_p()
-
-            prefix = str(_local_sample_path)
-            blobs = bucket.list_blobs(prefix=prefix)
-            for blob in blobs:
-                if not blob.name.endswith('/digest.json'):
-                    logging.info(f'Downloading blob: {blob.name}')
-                    blob.download_to_filename(blob.name)
-
-        samples = {
-            'dev_C1963.CHMI_CHMI3_Nex1': {
-                'total_jobs': 133,
-                'more_than_1_attempts': 12,
-                'more_than_2_attempts': 1,
-                'more_than_3_attempts': 0,
-                'cromwell_time_more_than_3_minutes_longer_total': 21,
-                'cromwell_time_more_than_4_minutes_longer_total': 7,
-                'cromwell_time_more_than_5_minutes_longer_total': 4,
-                'cromwell_time_more_than_6_minutes_longer_total': 2,
-                'cromwell_time_more_than_7_minutes_longer_total': 1,
-                'cromwell_time_more_than_8_minutes_longer_total': 0,
-                # insert more intelligent assertions here
-            }
-            # more samples if needed
-            # 'dev_C862.NA19238',
-            # 'dev_D5327.NA12878',
-            # 'dev_D5327.NA12891',
-            # 'dev_D5327.NA12892',
-            # 'dev_RP-1535.NA17-308'
-        }
+        # A cache of expensive-to-create GCS comparison paths.
+        gcs_comparison_path_by_subdir = {}
+        papi_versions = [VERSION_PAPI_V1, VERSION_PAPI_V2]
+
+        for papi_version in papi_versions:
+            subdir = subdir_for_papi_version(papi_version)
+            local_parent = ComparisonPath.create(subdir)
+
+            for sample_name in EXPECTATIONS.keys():
+                download_metadata_from_gcs_if_needed(sample_name, local_parent, bucket)
+                parents_to_test = [local_parent]
+                # Skip slow GCS testing if this environment variable is set.
+                if not os.environ.get('DIGESTER_TEST_LOCAL_ONLY'):
```

Comment:
TOL: Running unit tests against real GCS looks a bit like an overkill. Also is `gcs` mode the default one?

Reply:
I can make this opt-in rather than opt-out so it only runs in CI, unless asked for.

There isn't really a default mode, the `ComparisonPath`s treat path arguments as GCS or local as appropriate.

Reply:
>I can make this opt-in rather than opt-out so it only runs in CI, unless asked for.
That would be nice. Otherwise LGTM